test/unit_tests/test_braket_device.py

```diff
@@ -271,6 +297,30 @@ def test_batch_execute_non_parallel(monkeypatch):
         assert res == 1967
 
 
+@patch.object(AwsDevice, "run")
+def test_batch_execute_non_parallel_tracker(mock_run):
+    mock_run.return_value = TASK
+    dev = _aws_device(wires=2, foo="bar", parallel=False)
+
+    with QuantumTape() as circuit:
+        qml.Hadamard(wires=0)
+        qml.probs(wires=(0,))
+
+    callback = Mock()
+    with qml.Tracker(dev, callback=callback) as tracker:
+        dev.batch_execute([circuit])
```

Comment:
Could we please add a test with multiple circuits ? We should also compare the outputs to the parallel case and ensure that the values make sense.